### ISSUE TITLE
KinesisRecorder.submitAllRecords() loss data.

### ISSUE THREAD

**Issue by volic** [aws/aws-sdk-android#225]
submitAllRecords is long run method because it sends cached data to server through http connection.
When I invoke this method and internet connectivity is poor It runs more than a minute.
During this time the service had been killed by the android system and all sending data was lost.
When I checked the submitAllRecords method I noticed that you firstly remove data from cache and if it sends unsuccesfuly you return it back to the cache after some time.
Caching data is very sensetive operation and I cannot loose it during the killing of service.
Please remove cached records only when you will understand that the data send correctly or it is invalid.

``` 
public synchronized void submitAllRecords() {
 //...
   iterator.removeReadRecords();
 //...
   List<byte[]> failures = sender.sendBatch(streamName, data);
 //...
   saveRecord(bytes, streamName);
 //...
}
```

**Comment by wdane** [comment]
This is good feedback, this fix will be released in the next version of the Android SDK. However I cannot make a definite promise on when the next release will be.

**Comment by azsolinsky** [comment]
This was fixed in the 2.3.9 release.
